concurrent_test.go

```diff
@@ -347,27 +347,47 @@ func (w *worker) name() string {
 
 func (w *worker) run() (historyRecords, error) {
 	var rs historyRecords
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
 	for {
 		select {
 		case <-w.stopCh:
-			w.t.Logf("%q finished.", w.name())
 			return rs, nil
 		default:
 		}
 
-		op := w.pickOperation()
-		bucket, key := w.pickBucket(), w.pickKey()
-		rec, err := executeOperation(op, w.db, bucket, key, w.conf)
-		if err != nil {
-			readErr := fmt.Errorf("[%s: %s]: %w", w.name(), op, err)
-			w.t.Error(readErr)
-			w.errCh <- readErr
-			return rs, readErr
-		}
+		err := w.db.Update(func(tx *bolt.Tx) error {
+		inner:
+			for {
```

Comment:
that sounds like a fairly complicated way of running multiple operations. Why not just take a random(N) and for i/range instead?

Reply:
> that sounds like a fairly complicated way of running multiple operation

This is just the very simple usage of [BreakStmt = "break" [Label]](https://go.dev/ref/spec#Break_statements) syntax .

> Why not just take a random(N) and for i/range instead?

yes, `random(N)` is another option. I just want to execute as much operation as possible in each transaction instead of a randomized fixed count.

Reply:
Removed the label "inner" per @fuweid 's comment above.

Reply:
I was referring to the ticker logic, not the label. What if you have a very slow disk that only does one write every 5s? :)

Reply:
Variable environment is a good thing for test :) 

Also actually all the writing & deleting operation are in memory only (reading might be trigger loading page from disk), it should be fast enough.